process/factory/interceptorscontainer/baseInterceptorsContainerFactory.go

```diff
@@ -345,6 +345,16 @@ func (bicf *baseInterceptorsContainerFactory) generateMiniBlocksInterceptors() e
 	keys[noOfShards] = identifierMiniBlocks
 	interceptorsSlice[noOfShards] = interceptor
 
+	identifierAllShardsMiniBlocks := factory.MiniBlocksTopic + shardC.CommunicationIdentifier(core.AllShardId)
+
+	allShardsMiniBlocksInterceptorinterceptor, err := bicf.createOneMiniBlocksInterceptor(identifierAllShardsMiniBlocks)
```

Comment:
still do not know why all the shards has to know about all the miniblocks.

Reply:
all shards have to know about all peerMiniblocks. I could create just one miniblock and add all validatorInfos there.